gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/model/mapper/ServiceMapper.java

```diff
@@ -0,0 +1,140 @@
+        if (steps != null && !steps.isEmpty()) {
+            if (steps.size() > 1) {
+                log.error("Health check for " + stringToAppend + " cannot have more than one step ");
+                migrationResult.addIssue(
+                    new MigrationResult.Issue(
+                        "Health check for " + stringToAppend + " cannot have more than one step",
+                        MigrationResult.State.IMPOSSIBLE
+                    )
+                );
+            }
+            HealthCheckStep step = steps.get(0);
+            if (step.getResponse().getAssertions() != null && step.getResponse().getAssertions().size() > 1) {
+                log.error("Health check for " + stringToAppend + " cannot have more than one assertion");
+                migrationResult.addIssue(
+                    new MigrationResult.Issue(
+                        "Health check for " + stringToAppend + " cannot have more than one assertion",
+                        MigrationResult.State.IMPOSSIBLE
+                    )
+                );
+            }
+            try {
+                config.set(
+                    "headers",
+                    (step.getRequest().getHeaders() == null ? MAPPER.createArrayNode() : MAPPER.valueToTree(step.getRequest().getHeaders()))
+                );
+                config.set("method", MAPPER.valueToTree(step.getRequest().getMethod()));
+                config.set("target", MAPPER.valueToTree(step.getRequest().getPath()));
+                config.set("assertion", MAPPER.valueToTree(StringUtils.appendCurlyBraces(step.getResponse().getAssertions().get(0))));
+                config.set("overrideEndpointPath", MAPPER.valueToTree(step.getRequest().isFromRoot()));
+            } catch (Exception e) {
+                log.error("Unable to map configuration for Health check ", e);
+                return migrationResult.addIssue(
+                    new MigrationResult.Issue("Unable to map configuration for Health check", MigrationResult.State.IMPOSSIBLE)
+                );
+            }
+        }
+
+        // Build the v4 Service
+        return migrationResult.map(ser -> {
+            ser.setConfiguration(config.toString());
+            return ser;
+        });
+    }
+
+    private static MigrationResult<Service> convertEPHealthCheckService(
+        EndpointHealthCheckService v2EPHealthCheckService,
+        String type,
+        String name
+    ) {
+        MigrationResult<Service> plainHealthCheckServiceMigResult = convertHealthCheckService(v2EPHealthCheckService, type, name);
+        return plainHealthCheckServiceMigResult.map(plainHealthCheckService -> {
+            if (plainHealthCheckServiceMigResult.issues().isEmpty()) {
+                if (plainHealthCheckService != null) {
+                    plainHealthCheckService.setOverrideConfiguration(!v2EPHealthCheckService.isInherit());
+                }
+            }
+            return plainHealthCheckService;
+        });
```

Comment:
```suggestion
        return convertHealthCheckService(v2EPHealthCheckService, type, name)
            .map(plainHealthCheckService -> {
                plainHealthCheckService.setOverrideConfiguration(!v2EPHealthCheckService.isInherit());
                return plainHealthCheckService;
            });
```

MigrationResult manage issues for you